### PR TITLE
docs(general): Improve docs

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
Improve documentation related to go-libp2p-daemon building, and nim version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,18 +41,20 @@ Learn more about libp2p at [**libp2p.io**](https://libp2p.io) and follow libp2p'
...
 nimble install libp2p
 ```
 
 ## Getting Started
 You'll find the nim-libp2p documentation [here](https://status-im.github.io/nim-libp2p/docs/).
 
-**Go Daemon:**
-Please find the installation and usage intructions in [daemonapi.md](examples/go-daemon/daemonapi.md).
+### Go-Libp2p Daemon
+Remember you'll need to build the `go-libp2p-daemon` binary to use `nim-libp2p`.
```
Comment (diegomrsantos) [inline README.md:54]: I think `go-libp2p-daemon` is only necessary for running the tests, isn't it?

Comment (AlejandroCabeza) [reply README.md:54]: Is that so? Why is it needed to run the tests, then?

Comment (diegomrsantos) [reply README.md:54]: @Menduist told me this once:

> Why do we have testdaemon? Is it necessary?
Currently the only way we have to test pubsub interopability is using the go-daemon, that's why it's still here
Historically, nim-libp2p began has a simple go-daemon wrapper

Comment (AlejandroCabeza) [reply README.md]: Added a note for that.

File: examples/go-daemon/daemonapi.md [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+If everything goes correctly, the binary (`p2pd`) should be built and placed in the correct directory.
+If you find any issues, please head into our discord and ask for our asistance.
+
+After successfully building the binary, remember to add it to your path so it can be found. You can do that by running:
+```sh
+export PATH="$PATH:$HOME/go/bin"
+```
+> **Tip:** To make this change permanent, add the command above to your `.bashrc` file.
+
+## Manual
```
Comment (diegomrsantos) [inline examples/go-daemon/daemonapi.md:38]: Could we update the manual part as well? I'm not sure it is correct.

Comment (AlejandroCabeza) [reply examples/go-daemon/daemonapi.md:38]: Could you elaborate on what's wrong? Aside from the first block which relates to "installing nim-libp2p", that obviously doesn't belong here.
I haven't manually installed `go-libp2p-daemon`, so I don't know what's missing/wrong there.

Comment (diegomrsantos) [reply examples/go-daemon/daemonapi.md:38]: `git submodule update --init --recursive` seems it's saying nim-libp2p uses git submodule and I don't think it's true anymore.

`git checkout v0.0.1` I don't think that's the `go-libp2p-daemon` being used now.

Comment (AlejandroCabeza) [reply examples/go-daemon/daemonapi.md:38]: Wouldn't it be better then to just stick to the `script` based installation?

Comment (diegomrsantos) [reply examples/go-daemon/daemonapi.md:38]: I'm fine with removing the manual part.

**Review by lchenut — APPROVED** [review]

**Pull request merged**